tools/lint/src/main/java/com/grab/lint/LintAnalyzeCommand.kt

```diff
@@ -0,0 +1,43 @@
+package com.grab.lint
+
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.stream.Collectors
+import kotlin.io.path.isRegularFile
+import kotlin.io.path.name
+import kotlin.io.path.pathString
+import com.android.tools.lint.Main as LintCli
+
+class LintAnalyzeCommand : LintBaseCommand() {
+
+    override val createProjectXml: Boolean = true
+
+    override fun run(
+        workingDir: Path,
+        projectXml: File,
+        tmpBaseline: File,
+    ) {
+        val cliArgs = (defaultLintOptions + listOf(
+            "--cache-dir", workingDir.resolve("cache").pathString,
+            "--project", projectXml.toString(),
+            "--analyze-only" // Only do analyze
+        )).toTypedArray()
+        LintCli().run(cliArgs)
+        postProcessPartialResults()
+    }
+
+    private fun postProcessPartialResults() {
+        Files.walk(partialResults.toPath())
+            .filter { it.isRegularFile() }
+            .collect(Collectors.toList())
+            .parallelStream()
+            .forEach { path ->
+                if ("lint-definite-all.xml" in path.name) {
+                    Files.delete(path)
+                } else {
+                    sanitizer.sanitize(path.toFile())
```

Comment:
I think should be fine no? Since as far I see it does not have any state and should be okay to access across threads.

Reply:
From what I know this can cause issue, not all the time but it can, since the most intensive action in this method is sanitizer so it make sense to me to have different object for different threads.